packages/block-editor/src/hooks/utils.js

```diff
@@ -16,22 +16,28 @@ import { useSetting } from '../components';
 import { useSettingsForBlockElement } from '../components/global-styles/hooks';
 
 /**
- * Removed falsy values from nested object.
+ * Removed falsy values from nested object. If `allowZeros` is `true`, then
+ * zero values will be allowed within the resulting object.
+ *
+ * @param {*}       object             Object to be cleaned.
+ * @param {Object}  options            Options config.
+ * @param {boolean} options.allowZeros Whether to allow zero values.
  *
- * @param {*} object
  * @return {*} Object cleaned from falsy values
  */
-export const cleanEmptyObject = ( object ) => {
+export const cleanEmptyObject = ( object, { allowZeros } = {} ) => {
 	if (
 		object === null ||
 		typeof object !== 'object' ||
 		Array.isArray( object )
 	) {
 		return object;
 	}
+	const recursiveCleanObject = ( nestedObject ) =>
+		cleanEmptyObject( nestedObject, { allowZeros } );
```

Comment:
It seems more work to destructure the options (and involves creating a new object on each recursion), so not destructuring might be tidier:
```suggestion
export const cleanEmptyObject = ( object, options ) => {
	if (
		object === null ||
		typeof object !== 'object' ||
		Array.isArray( object )
	) {
		return object;
	}
	const recursiveCleanObject = ( nestedObject ) =>
		cleanEmptyObject( nestedObject, options );
```